src/shared/lib/comparison/ComparisonStore.ts

```diff
@@ -557,100 +558,157 @@ export default class ComparisonStore {
-                molecularProfiles,
-                molecularProfile => molecularProfile.studyId
-            );
-            if (this.activeGroups.result!.length > 1) {
-                let groups: MolecularProfileCasesGroupFilter[] = _.map(
-                    this.activeGroups.result,
-                    group => {
-                        const molecularProfileCaseIdentifiers = _.flatMap(
-                            group.studies,
-                            study => {
-                                return _.map(
-                                    this.usePatientLevelEnrichments
-                                        ? study.patients
-                                        : study.samples,
-                                    caseId => {
-                                        return {
-                                            caseId,
-                                            molecularProfileId:
-                                                studyMolecularProfileMap[
-                                                    study.id
-                                                ].molecularProfileId,
-                                        };
-                                    }
-                                );
-                            }
+        invoke: () => {
+            return Promise.resolve(
+                _.chain(this.enrichmentAnalysisGroups.result)
+                    .map(group => {
+                        const filteredSamples = group.samples.filter(
+                            sample =>
+                                this.selectedStudyMutationEnrichmentProfileMap
+                                    .result![sample.studyId] !== undefined
                         );
                         return {
-                            name: group.nameWithOrdinal,
-                            molecularProfileCaseIdentifiers,
+                            ...group,
+                            count: filteredSamples.length,
+                            samples: filteredSamples,
+                            description: `Number (percentage) of ${
+                                this.usePatientLevelEnrichments
+                                    ? 'patients'
+                                    : 'samples'
+                            } in ${
+                                group.name
+                            } that have a mutation in the listed gene.`,
                         };
-                    }
-                );
+                    })
+                    .filter(group => group.samples.length > 0)
+                    .value()
+            );
+        },
+    });
+
+    readonly mutationEnrichmentDataRequestGroups = remoteData({
+        await: () => [
+            this.mutationEnrichmentAnalysisGroups,
+            this.selectedStudyMutationEnrichmentProfileMap,
+        ],
+        invoke: async () => {
+            let groups: MolecularProfileCasesGroupFilter[] = _.chain(
```

Comment:
chaining with these big statements makes things kind of hard to read.  granted prettier tends to bloat everything vertically.   
i would either separate the filter statement ```const nonEmptyGroups = groups.filter(..)``` or use reduce to accomplish the filtering

Reply:
fyi, we don't need lodash here because map, reduce, filter are available natively on array now (and chaineable)